components/layout/display_list_builder.rs

```diff
@@ -60,6 +60,10 @@ use table_cell::CollapsedBordersForCell;
 use url::Url;
 use util::opts;
 
+fn get_cyclic<T>(arr: &[T], index: usize) -> &T {
```

Comment:
You should be able to use `get_{prop}_mod(index)` instead of this hand-rolled thing.